ingestion/src/metadata/orm_profiler/profiler/sampler.py

```diff
@@ -67,8 +67,13 @@ def get_sample_query(self) -> Query:
                 )
                 .cte(f"{self.table.__tablename__}_rnd")
             )
+        table_query = self.session.query(self.table)
         return (
-            self.session.query(self.table)
+            self.session.query(
+                self.table,
+                (ModuloFn(RandomNumFn(), table_query.count())).label(RANDOM_LABEL),
+            )
+            .order_by(RANDOM_LABEL)
```

Comment:
How would that work for row sampling?

Reply:
self.session.query(
                self.table,
                (ModuloFn(RandomNumFn(), table_query.count())).label(RANDOM_LABEL),
            )
            .order_by(RANDOM_LABEL)

this will help with random row sampling, 
everytime random number is generated for rows,
it get’s ordered by RANDOM_LABEL col that was generated.

it will be something like:

        row1, 4
        row2, 1
        row3, 2
        row4, 3

if we add profile sample as 3
it gets ordered like this:

        row2, 1
        row3, 2
        row4, 3
        row1, 4

then we apply limit i.e. 3

        row2,1
        row3,2
        row4,3

and post that,
we have rnd.c.random <= 3
sampled = session_query.where(rnd.c.random <= self.profile_sample).cte(
            f"{self.table.__tablename__}_sample"
        )
the sampled cte gets created with the above 3 rows for profile sampling

Reply:
Wouldn't it change the behavior of the sampling by percent though? Sorry if I am missing something here.

Reply:
Sampling with percentage is taken care of from Line 59 - 69